### PR TITLE
Lollipop rework

### PR DESCRIPTION
**Pull request by FluffTheMoth01**
## About The Pull Request
So i forgot to remerge my last one so here we are again!
Don't you hate how you have to CRONCH your lollipops? Don't you hate how you can't SUCC? Well now you can! The lollipop can now be put in a mask slot and SUCCED! ~~Also lollipops are in vending machines.~~

## Why It's Good For The Game
Fixes muh immersion.

## Changelog
:cl:
tweak: Lollipops can now be succed by putting them in a mask slot.
/:cl:

### CODE REVIEW

**Comment by HomocidlePancakes** [comment]
Lolipops are cigarettes now. Cool.

File: code/modules/vending/snack.dm [hunk 1]
```diff
@@ -12,6 +12,7 @@
 					/obj/item/reagent_containers/food/snacks/no_raisin = 6,
 					/obj/item/reagent_containers/food/drinks/dry_ramen = 3,
 					/obj/item/reagent_containers/food/snacks/energybar = 6)
+					/obj/item/reagent_containers/food/snacks/lollipop = 6)
```
Comment (nemvar) [inline code/modules/vending/snack.dm]: ebic powercreep

**Comment by anconfuzedrock** [comment]
pops heal, shouldn't be in vending machines.

**Comment by Dawson1917** [comment]
once again, a feature inaccessible to clowns and mimes... when will it end?

**Comment by partykp** [comment]
Really shouldnt be in vendings imo but otherwise its a great feature

**Comment by fludd12** [comment]
### _**SUCC**_

**Comment by Shadowflame909** [comment]
If in a vendor. Should be premium.

**Comment by SpaceManiac** [comment]
doesn't compile
```
code/modules/vending/snack.dm:15:error: inconsistent indentation
code/modules/vending/snack.dm:15:error: ): expected }
code/modules/vending/snack.dm:2:error: location of top-most unmatched {
```

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
doesn't compile
```
code/modules/vending/snack.dm:15:error: inconsistent indentation
code/modules/vending/snack.dm:15:error: ): expected }
code/modules/vending/snack.dm:2:error: location of top-most unmatched {
```

**Comment by IllotAFV** [comment]
Medical cyborgs, if aiming mouth, should be able to forcefully put on lollipops on someone, when throwing them. Of course, if mask slot of that someone is empty.

**Comment by SpacePrius** [comment]
will fix the compile issue tomorrow when I get home. Staying at my cousins
for Saturday.

**Comment by SpacePrius** [comment]
decided to fix it now.

**Comment by Qustinnus** [comment]
i want to rework food into a component and things that make food code weirder like this hurt my soul :(

**Comment by SpacePrius** [comment]
it's a subtype of food, any changes to food still apply to this, so it
shouldn't make your food changes much harder.

**Comment by Qustinnus** [comment]
I'm making food a component so I'm removing the entire class :(

**Comment by Dawson1917** [comment]
There's a TC-purchaseable pack of syndicate smokes that has the advertised benefit of containing omnizine for a passive heal while smoked (it's pretty shitty and slow but it's something). That should probably be buffed to not be so weak if lollipops now do the exact same thing

**Comment by SpacePrius** [comment]
lollipops always had omnizine

**Comment by Dawson1917** [comment]
> 
> 
> lollipops always had omnizine

Yes but now they can be stuck in your mask slot for a passive heal in the exact same way as the syndicate smokes. Not sure if the rate of reagent transfer is faster or slower than smoking but a free and limitless item from mediborgs will now do the same job as a 1tc uplink item, hopefully someone will end up making the smokes more useful in future

**Comment by Shadowflame909** [comment]
Omnizine is getting some PR changes as well.

I'd probably replace the syndi-smokes with some stimulants.

**Comment by dee-dubya** [comment]
Give us the S U C C

**Comment by actioninja** [comment]
succ granted

**Pull request merged**